astropy/io/misc/asdf/tags/coordinates/tests/test_angle.py

```diff
@@ -3,7 +3,7 @@
 
 import pytest
 
-from astropy import __minimum_asdf_version__
+from astropy.io.misc.asdf.tags.helpers import __minimum_asdf_version__
 
 asdf = pytest.importorskip('asdf', minversion=__minimum_asdf_version__)
```

Comment:
Why do we need `minversion=__minimum_asdf_version__` ? Since this is the minimum version installed by pip the minversion requirement should always be met.
I also wonder if we could skip all the asdf tests with a ` astropy/io/misc/asdf/conftest.py` file, if asdf is not installed.

Reply:
I didn't write this. Maybe @eslavich or @jdavies-st can advise?

Reply:
> Why do we need `minversion=__minimum_asdf_version__` ?

The `asdf` package is in the extras, so it is possible to install astropy into an environment with an incompatible version of asdf and not see a warning.  But that's a weird enough edge case that I don't think we need to handle it in the code.

We don't consistently set `minversion`, for example:

https://github.com/astropy/astropy/blob/v4.3.dev/astropy/io/misc/asdf/tags/coordinates/tests/test_skycoord.py#L11

> I also wonder if we could skip all the asdf tests with a `astropy/io/misc/asdf/conftest.py` file, if asdf is not installed.

This seems way better than adding `pytest.importorskip` to every test file.

Reply:
Yes ok. Users should have dependencies that met our requirements. For tests we ask people to `pip install -e .[tests]` so having an older version of whatever dependency should not happen. And I don't think we need to add dozens lines of code to make sure it doesn't happen. It's the same for other dependencies, we cannot check everywhere that some version is not recent enough.

Reply:
And btw, that's why tox is so important. It will make sure that the correct versions are installed before running tests :).

Reply:
I couldn't find any precedence to skip all tests for a subpackage in `conftest.py` here, so I went the less controversial route of keeping the skip in the module but removing minversion check.

Reply:
It seems that tests are already skipped if entry points are not defined, e.g.:
https://github.com/astropy/astropy/blob/master/astropy/io/misc/asdf/tests/__init__.py
I'm not sure if there is a difference between the entry points not installed and asdf itself not installed, but I would guess that tests are already skipped if asdf is not installed, without the `pytest.importorskip` calls.

This is repeated in various directories, so I guess a conftest file in the root asdf module with something inspired from https://stackoverflow.com/a/47700320/4728722 would allow to skip all the tests.

Reply:
Actually `pytest.importorskip('asdf')` also does the import, so it's fine to keep it, and it's needed for collecting for tests.